### ISSUE TITLE
(assets): force re-generation of assets via the CLI

### ISSUE THREAD

**Issue by nija-at** [aws/aws-cdk#12800]
We hit a [bug](https://github.com/aws/aws-cdk/issues/12536) in NodeJS version that generated corrupted assets. 

The problem is that just changing to a non-buggy version of NodeJS is insufficient. This is because the assets are cached in the `cdk.out/` directory and cached in the staging S3 bucket.
To resolve this issue, both has to be manually deleted. 

This is a request to provide an option via the CLI where all asset caches are ignored and asset packaging/staging is re-run.

---

This is a :rocket: Feature Request

**Comment by jk2l** [comment]
I would like to re-open this too. I encountered this bug in prod environment https://github.com/aws/aws-cdk/issues/30967 

One of our developer done a prod release via --exclusive flag, this caused the asset bundled without dependency, our lambda was completely broken, because of the CDK ignore asset upload if it already exist. Both of CodePipeline or local CDK cli won't fix the issue.

At the end I need to manually go into S3 bucket hunt down each one of those lambda zip file in the staging asset bucket, and delete them manually. This is not really an ideal situation, lucky we only affected very few lambda during this deployment, but imagine if i have a deployment that trigger lot of lambda this will be a nightmare to fix (e.g. i recently deployed 1000 lambda update accross 70 accounts, because Python3.7 retired with EOL)
